engine/runtime/src/main/java/org/enso/interpreter/runtime/error/WithWarnings.java

```diff
@@ -43,36 +56,29 @@ public Object getValue() {
   }
 
   public WithWarnings append(Warning... newWarnings) {
-    return new WithWarnings(value, warnings.append(newWarnings));
+    return new WithWarnings(value, warnings, newWarnings);
   }
 
-  public WithWarnings append(ArrayRope<Warning> newWarnings) {
-    return new WithWarnings(value, warnings.append(newWarnings));
+  public WithWarnings prepend(ArrayRope<Warning> newWarnings) {
```

Comment:
I do wonder whether we should have `append` *and* `prepend` still. They don't really make sense now given that we mostly rely on the creation time.